networkx/algorithms/euler.py

```diff
@@ -15,10 +21,15 @@
 
 from ..utils import arbitrary_element
 
+from collections import deque
+
 __author__ = """\n""".join(['Nima Mohammadi (nima.irt[AT]gmail.com)',
-                            'Aric Hagberg <hagberg@lanl.gov>'])
+                            'Aric Hagberg <hagberg@lanl.gov>',
```

Comment:
This is why I suggested #1675 (which I consider not really a "Documentation" issue, but a "Defect", for the problems you propose here). This is how, for example, Python Software Foundation does it.

The way it roughly seems to have been done until now is this: a contributor distributes the code that he or she authors (and on which he or she has a copyright) to NetworkX under an appropriate license (i.e. the Modified BSD License), and the copyright for that bit of contributed code (i.e. the author and year) is listed.